File: scripts/debug/auth/debug_test.py
#!/usr/bin/env python3
"""Debug script to test http://localhost:3000/ and capture errors."""

import asyncio
from playwright.async_api import async_playwright, Page
import json
import sys

async def test_app():
    async with async_playwright() as p:
        browser = await p.chromium.launch()
        page = await browser.new_page()

        # Capture console messages
        console_messages = []
        page.on("console", lambda msg: console_messages.append({
            "type": msg.type,
            "text": msg.text
        }))

        # Capture network requests/responses
        network_errors = []
        async def on_response(response):
            if response.status >= 400:
                try:
                    body = await response.text()
                    network_errors.append({
                        "url": response.url,
                        "status": response.status,
                        "body": body[:500]  # First 500 chars
                    })
                except:
                    pass

        page.on("response", on_response)

        # Navigate to app
        print("🔄 Loading http://localhost:3000/...")
        try:
            await page.goto("http://localhost:3000/", wait_until="networkidle")
            print("✅ Page loaded successfully")
        except Exception as e:
            print(f"❌ Navigation failed: {e}")
            await browser.close()
            return

        # Wait a bit for any JS errors
        await asyncio.sleep(2)

        # Take screenshot
        await page.screenshot(path="screenshot.png")
        print("📸 Screenshot saved: screenshot.png")

        # Get page content
        content = await page.content()

        # Check for error indicators
        error_indicators = [
            "HTTPError",
            "500",
            "error",
            "Internal Server Error",
            "RunnerError"
        ]

        has_errors = any(indicator in content for indicator in error_indicators)

        print("\n📋 Debug Info:")
        print(f"   Page title: {await page.title()}")
        print(f"   URL: {page.url}")

        if console_messages:
            print(f"\n🖨️  Console Messages ({len(console_messages)}):")
            for msg in console_messages[-10:]:  # Last 10
                print(f"   [{msg['type']}] {msg['text'][:100]}")

        if network_errors:
            print(f"\n⚠️  Network Errors ({len(network_errors)}):")
            for err in network_errors:
                print(f"   {err['status']} {err['url']}")
                print(f"   Body: {err['body']}\n")

        if has_errors:
            print(f"\n❌ ERROR INDICATORS FOUND IN HTML")
            # Extract error section
            if "HTTPError" in content:
                idx = content.find("HTTPError")
                print(f"   Context: ...{content[max(0, idx-100):idx+200]}...")
        else:
            print(f"\n✅ No obvious error indicators found")

        await browser.close()

if __name__ == "__main__":
    asyncio.run(test_app())

File: scripts/debug/auth/debug_test.ts
import { chromium, type ConsoleMessage, type Response } from 'playwright';

interface CapturedConsoleMessage {
  type: string;
  text: string;
}

interface NetworkError {
  url: string;
  status: number;
  body: string;
}

const responseHasFailureStatus = (response: Response): boolean => response.status() >= 400;

const toConsoleMessage = (message: ConsoleMessage): CapturedConsoleMessage => ({
  type: message.type(),
  text: message.text(),
});

const run = async (): Promise<void> => {
  const browser = await chromium.launch();
  const page = await browser.newPage();

  const consoleMessages: CapturedConsoleMessage[] = [];
  const networkErrors: NetworkError[] = [];

  page.on('console', (message) => {
    consoleMessages.push(toConsoleMessage(message));
  });

  page.on('response', async (response) => {
    if (!responseHasFailureStatus(response)) {
      return;
    }

    try {
      const body = await response.text();
      networkErrors.push({
        url: response.url(),
        status: response.status(),
        body: body.slice(0, 500),
      });
    } catch {
      // Ignore unreadable response bodies in this debug script.
    }
  });

  console.log('Loading http://localhost:3000/...');

  try {
    await page.goto('http://localhost:3000/', { waitUntil: 'networkidle' });
    console.log('Page loaded successfully');
  } catch (error: unknown) {
    console.error(`Navigation failed: ${error instanceof Error ? error.message : String(error)}`);
    await browser.close();
    return;
  }

  await page.waitForTimeout(2000);
  await page.screenshot({ path: 'screenshot.png' });
  console.log('Screenshot saved: screenshot.png');

  const content = await page.content();
  const errorIndicators = ['HTTPError', '500', 'error', 'Internal Server Error', 'RunnerError'];
  const hasErrors = errorIndicators.some((indicator) => content.includes(indicator));

  console.log('\nDebug Info:');
  console.log(`  Page title: ${await page.title()}`);
  console.log(`  URL: ${page.url()}`);

  if (consoleMessages.length > 0) {
    console.log(`\nConsole Messages (${consoleMessages.length}):`);
    for (const message of consoleMessages.slice(-10)) {
      console.log(`  [${message.type}] ${message.text.slice(0, 100)}`);
    }
  }

  if (networkErrors.length > 0) {
    console.log(`\nNetwork Errors (${networkErrors.length}):`);
    for (const error of networkErrors) {
      console.log(`  ${error.status} ${error.url}`);
      console.log(`  Body: ${error.body}\n`);
    }
  }

  if (hasErrors) {
    console.log('\nError indicators found in HTML');
    const index = content.indexOf('HTTPError');
    if (index >= 0) {
      console.log(`  Context: ...${content.slice(Math.max(0, index - 100), index + 200)}...`);
    }
  } else {
    console.log('\nNo obvious error indicators found');
  }

  await browser.close();
};

void run();
